### ISSUE TITLE
After refreshing, user is logged out

### ISSUE THREAD

**Issue by pg7812** [Saka7/spring-boot-angular4-boilerplate#3]
auth.service.ts::refresh
```
  public refresh() {
    this.logout();
    const user = sessionStorage.getItem('user');
    if (user) {
      this.saveUserDetails(JSON.parse(user));
    }
  }
```

Is this ok, or const user and logout should be replaced?
Even so, the token is lost, after refreshing.

**Comment by pg7812** [comment]
Why it is not just : 
```
  public refresh() {
    const user = sessionStorage.getItem('user');
    if (user) {
      this.saveUserDetails(JSON.parse(sessionStorage.getItem('user')));
    }
  }
```

**Comment by Saka7** [comment]
Nice observation, loosing token after every `authService` re-creation isn't acceptable.
Initially, I've made `refresh` method only for testing purposes.

As in the quick-start only one `ngModule` is used and `authService` is provided module-wide I don't even notice token lost. But, if multiple modules are used or service is provided component-wide, current implementation of `refresh` method will cause some unexpected behavior.

Thanks, for reporting an issue. I'll fix it as soon as I have some free time.

**Comment by Saka7** [comment]
Fixed in 982830cce31fc2c9b1bee4e01a6548a11aeaa27e.
